CMakeLists.txt

```diff
@@ -176,12 +172,121 @@ else()
     find_path(Protobuf_INCLUDE_DIR google/protobuf/service.h
               PATHS ${_PROTOBUF_INSTALL_PREFIX}/include
               NO_DEFAULT_PATH)
-    find_package(Protobuf REQUIRED)
+    find_package(Protobuf)
+    if (Protobuf_FOUND)
+      if (ONNX_USE_PROTOBUF_V22 AND "${Protobuf_VERSION}" VERSION_LESS "4.22.0")
```

Comment:
FetchContent won't set that variable, so I set one. If protobuf is an external lib, it will have the version.
For our url used as external lib, the version is "4.22.3", the old protobuf is "3.21.12".
![image](https://github.com/onnx/onnx/assets/19584326/0c0ca81e-a268-49bf-9798-557074b85c4a)